ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java

```diff
@@ -408,14 +394,22 @@ private Widget getCollapseBtn(ContentPanel panel) {
     @Override
     public void updateMetadataFromTemplateView(List<Avu> metadataList) {
         userMdGrid.mask();
+        if(userMdListStore.size() == 0) {
+           userMdListStore.addAll(metadataList);
+           userMdGrid.unmask();
+           return;
+        }
+        List<Avu> itemsToAdd = new ArrayList<>();
         for (Avu md : metadataList) {
-            for (Avu storemd : userMdListStore.getAll()) {
-                if (storemd.getAttribute().equals(md.getAttribute())) {
-                    storemd.setValue(md.getValue());
-                    userMdListStore.update(storemd);
-                }
+            Avu model = userMdListStore.findModel(md);
+            if (model != null) {
+                model.setValue(md.getValue());
+                userMdListStore.update(model);
+            } else {
+                itemsToAdd.add(presenter.setAvuModelKey(md));
```

Comment:
If you decide to go the route I mentioned earlier, then a ModelKeyProvider for userMdListStore could also do the same get/set routine.